nbodykit/utils/config.py

```diff
@@ -65,31 +65,32 @@ def fill_namespace(ns, arg, config, missing):
         a list to add arguments that are missing, i.e., required and not
         present in the input configuration files
     """
-    # the name of the parameter
-    name = arg.name.split('.')[-1]
+    # the name of the parameter (as taken from the schema)
+    schema_name = arg.name.split('.')[-1]
     
     # no subfields
     if not len(arg.subfields): 
     
-        if config is not None and name.lower() in [k.lower() for k in config]:
-            value = config.pop(name)
+        # check if the schema argument is present in configuration file
+        if config is not None and schema_name.lower() in [k.lower() for k in config]:
```

Comment:
and the keys of the ``config`` dict here are the names of the parameters from the configuration file